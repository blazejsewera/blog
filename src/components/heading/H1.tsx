import type { Component } from 'solid-js'
import { cx } from '../util/classes'
import { BIG_TITLE_FONT, HeadingProps } from './common'

export const H1: Component<HeadingProps> = ({ children }) => (
  <h1 class={cx(BIG_TITLE_FONT, 'text-4xl')}>{children}</h1>
)
