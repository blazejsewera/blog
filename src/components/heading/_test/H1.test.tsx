import { H1 } from '../H1'
import { render } from '@solidjs/testing-library'

describe('<H1 />', () => {
  it('is accessible as first-level header', async () => {
    const { findByRole, unmount } = render(() => <H1>Test</H1>)
    const h1 = await findByRole('heading', { level: 1 })
    expect(h1).toBeInTheDocument()
    unmount()
  })

  it('has proper text', async () => {
    const expected = 'Test'
    const { findByText, unmount } = render(() => <H1>{expected}</H1>)
    const actual = await findByText(expected)
    expect(actual).toBeInTheDocument()
    unmount()
  })

  it('is properly styled', () => {
    const { container, unmount } = render(() => <H1>Test</H1>)
    expect(container).toMatchSnapshot()
    unmount()
  })
})
