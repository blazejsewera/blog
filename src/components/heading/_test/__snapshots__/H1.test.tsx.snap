// Vitest Snapshot v1

exports[`<H1 /> > is properly styled 1`] = `
<div>
  <h1
    class="font-geometric text-4xl"
  >
    Test
  </h1>
</div>
`;
